services/QuillLMS/app/models/user_pack_sequence_item.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: user_pack_sequence_items
+#
+#  id                    :bigint           not null, primary key
+#  status                :string           default("locked")
+#  created_at            :datetime         not null
+#  updated_at            :datetime         not null
+#  pack_sequence_item_id :bigint
+#  user_id               :bigint
+#
+# Indexes
+#
+#  index_user_pack_sequence_items_on_pack_sequence_item_id  (pack_sequence_item_id)
+#  index_user_pack_sequence_items_on_user_id                (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (pack_sequence_item_id => pack_sequence_items.id)
+#  fk_rails_...  (user_id => users.id)
+#
+class UserPackSequenceItem < ApplicationRecord
+  STATUSES = [
+    LOCKED = 'locked',
+    UNLOCKED = 'unlocked'
+  ]
+
+  belongs_to :user
+  belongs_to :pack_sequence_item
+
+  validates :status, inclusion: { in: STATUSES }
```

Comment:
Can `user_id` or `pack_sequence_item_id` ever be null for a valid saved object?

If not, consider `validates presence` for them and making the DB fields non-null.

Reply:
Good point.  I mistakenly assumed `foreign_key: true` prevented nullable references.